src/Shell/ProcessFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CloudPatches\Shell;
+
+use Magento\CloudPatches\Filesystem\DirectoryList;
+use Symfony\Component\Process\Process;
+
+/**
+ * Factory method for Process.
+ *
+ * @see Process
+ */
+class ProcessFactory
+{
+    /**
+     * @var DirectoryList
+     */
+    private $directoryList;
+
+    /**
+     * @param DirectoryList $directoryList
+     */
+    public function __construct(DirectoryList $directoryList)
+    {
+        $this->directoryList = $directoryList;
+    }
+
+    /**
+     * @param array $cmd
+     * @return Process
+     */
+    public function create(array $cmd): Process
+    {
+        return new Process(
```

Comment:
why commands are imploded by space?

Reply:
It is BC. Starting from Symfony/Process 4.2 it accepts only array

Reply:
then, maybe retrieve only first element and throw exception if there are more elements in array?

Reply:
That's the same method which Symfony does to preserve compatibility